### ISSUE TITLE
compilation errors for tcp::socket

### ISSUE THREAD

**Issue by johannesmayer** [CCPPETMR/SIRF#200]
@evgueni-ovtchinnikov,  we're getting errors like
````
/home/sirfuser/devel/JohSIRF/Build/sources/SIRF/src/xGadgetron/cGadgetron/gadgetron_client.h:104:20: error: ‘tcp’ has not been declared
  virtual void read(tcp::socket* s) = 0;
````
Related line is https://github.com/CCPPETMR/SIRF/blob/bdcd47ac8623657d0e4d4710713b02e1f6071178/src/xGadgetron/cGadgetron/gadgetron_client.h#L106

Is this since your `namespace stir` commits?

**Comment by johannesmayer** [comment]
needs `boost::asio::ip::tcp::socket`

**Comment by evgueni-ovtchinnikov** [comment]
yes, now you need to have these lines in your source before you can include `xgadgetron` stuff:

    #include <boost/asio.hpp>
    using boost::asio::ip::tcp;

they used to be in `gadgetron_client.h`, which was a bad thing, as Kris said

**Comment by johannesmayer** [comment]
@KrisThielemans says that we need the `#include` in `gadgetron_client.h` (it's already there) but there it needs to explicitly use `boost::asio::ip::tcp::socket`. we cannot rely on people adding `using` statements before including a SIRF file.

**Comment by evgueni-ovtchinnikov** [comment]
ok, restored

**Comment by evgueni-ovtchinnikov** [comment]
check if works ok and close, please

**Comment by KrisThielemans** [comment]
sorry for misunderstanding.

as before, we shouldn't have `using boost:...;` (importing boost stuff into the global namespace) in our .h files. So, in `gadgetron_client.h`, please remove that statement, and use the full `boost::asio::ip::tcp::socket`. 

ugly, I know.

**Comment by evgueni-ovtchinnikov** [comment]
error C2885: 'boost::asio::ip::tcp::socket': not a valid using-declaration at non-class scope

without `socket` works ok

**Comment by KrisThielemans** [comment]
I'm confused... don't use `using` but simply
```
virtual void read(boost::asio::ip::tcp::socket* s) = 0; 
```
(and similar wherever you use `tcp::socket`). if that's too ugly, I guess you could add a public typedef inside the class
```
typedef boost::asio::ip::tcp::socket socket;
```
and just use `socket` inside the .h file. But then you'd really have to use `GadgetronClientMessageReader ::socket` in the .cpp. Probably not worth it.
